src/PHPSQLParser/processors/UnionProcessor.php

```diff
@@ -119,22 +119,27 @@ protected function splitUnionRemainder($queries, $unionType, $outputArray)
         $finalQuery = [];
 
         //If this token contains a matching pair of brackets at the start and end, use it as the final query
-        foreach ($outputArray as $key => $token) {
-            $tokenAsArray = str_split(trim($token));
-            $keyCount = max(array_keys($tokenAsArray));
-
-            if (($tokenAsArray[0] == '(' && $tokenAsArray[$keyCount] == ')')) {
```

Comment:
The error was here. This code assumes there is only one token (the whole SELECT query) but it actually catches any token that has parenthesis in the whole query.